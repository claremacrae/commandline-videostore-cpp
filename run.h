#ifndef COMMANDLINE_VIDEOSTORE_CPP_RUN_H
#define COMMANDLINE_VIDEOSTORE_CPP_RUN_H
#include <iosfwd>
#include <string>

void run(std::istream& in, std::ostream& out);

void calculateAmountAndPoints(const std::string& category,
                              int daysRented,
                              double& thisAmount,
                              int& frequentRenterPoints);

#endif
