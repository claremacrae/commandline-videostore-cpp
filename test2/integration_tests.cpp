#include "run.h"
#include "catch2/catch.hpp"
#include "ApprovalTests.hpp"
#include <fmt/core.h>
using namespace ApprovalTests;

#include <sstream>

TEST_CASE("test empty input") {
  std::stringstream in;
  std::stringstream out;
  run(in, out);
  Approvals::verify(out.str());
}

TEST_CASE("test real input") {
  std::stringstream in;
  in << "Clare Macrae\n";
  in << "1 2\n";
  in << "1 3\n";
  std::stringstream out;
  run(in, out);
  Approvals::verify(out.str());
}

TEST_CASE("test non-production data") {
  std::vector<std::string> inputs1{"REGULAR", "CHILDRENS", "NEW_RELEASE"};
  std::vector<int> inputs2{1, 2, 3, 4};
  CombinationApprovals::verifyAllCombinations(
      "(Category, days)",
      [&](auto category, auto days) {
        double amount = 0;
        int points = 0;
        calculateAmountAndPoints(category, days, amount, points);
        return fmt::format("\t(Amount: {:7.2f}\tPoints: {}", amount, points);
      },
      inputs1,
      inputs2);
}
