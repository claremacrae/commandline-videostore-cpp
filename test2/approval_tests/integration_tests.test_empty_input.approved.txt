0: The Shawshank Redemption (1994)
1: The Godfather (1972)
2: The Godfather: Part II (1974)
3: The Dark Knight (2008)
4: Pulp Fiction (1994)
5: The Good, the Bad and the Ugly (1966)
6: The Lord of the Rings: The Return of the King (2003)
7: Fight Club (1999)
8: The Lord of the Rings: The Fellowship of the Ring (2001)
9: Interstellar (2014)
10: Whiplash (2014)
11: Birdman (2014)
12: Up (2009)
13: WALLÂ·E (2008)
Enter customer name: Choose movie by number followed by rental days, just ENTER for bill:
Rental Record for 
You owed 0.0
You earned 0 frequent renter points

